### PR TITLE
fix: ICA export genesis patch

### PR DESCRIPTION
**Pull request by czarcas7ic**
## What is the purpose of the change

During v11 I had a custom branch I and others used to get passed an ICA bug when doing a state export. This implements the fix in our v11 sdk fork which will then be backported to v11x (and maybe a minor release?) so that if someone needs to state export v11 height in the future they are able to.

### CODE REVIEW

File: x/auth/types/genesis.go [hunk 1]
```diff
@@ -33,6 +33,9 @@ func NewGenesisState(params Params, accounts GenesisAccounts) *GenesisState {
 func (g GenesisState) UnpackInterfaces(unpacker types.AnyUnpacker) error {
 	for _, any := range g.Accounts {
 		var account GenesisAccount
+		if any.TypeUrl == "/ibc.applications.interchain_accounts.v1.InterchainAccount" {
```
Comment (alexanderbez) [inline x/auth/types/genesis.go:36]: woaaa, why is this necessary?

**Review by alexanderbez — APPROVED** [review]

File: x/auth/types/genesis.go (hunk 1 above)
Comment (alexanderbez) [reply x/auth/types/genesis.go:36]: Also might be better to call MsgType on the type directly instead of a hard-coded string in case it changes.

Comment (czarcas7ic) [reply x/auth/types/genesis.go:36]: we didn't implement ICA correctly in v11, so at some height when someone on mainnet created an ICA it errors the export

Comment (czarcas7ic) [reply x/auth/types/genesis.go:36]: IRT the type vs hard code, this is no longer needed in v12 so even if this changed in the future it would not affect this AFAIU!
